packages/scoped-custom-element-registry/src/scoped-custom-element-registry.ts

```diff
@@ -120,756 +120,789 @@ type ParametersOf<
-  >();
-  private readonly _awaitingUpgrade = new Map<string, Set<HTMLElement>>();
+  // TBD: This part of the spec proposal is unclear:
+  // > Another option for looking up registries is to store an element's
+  // > originating registry with the element. The Chrome DOM team was concerned
+  // > about the small additional memory overhead on all elements. Looking up the
+  // > root avoids this.
+  const scopeForElement = new WeakMap<Node, Element | ShadowRoot>();
+
+  class AsyncInfo<T> {
+    readonly promise: Promise<T>;
+    readonly resolve: (val: T) => void;
+    constructor() {
+      let resolve: (val: T) => void;
+      this.promise = new Promise<T>((r) => {
+        resolve = r;
+      });
+      this.resolve = resolve!;
+    }
+  }
 
-  define(tagName: string, elementClass: CustomElementConstructor) {
-    tagName = tagName.toLowerCase();
-    if (this._getDefinition(tagName) !== undefined) {
-      throw new DOMException(
-        `Failed to execute 'define' on 'CustomElementRegistry': the name "${tagName}" has already been used with this registry`
+  // Constructable CE registry class, which uses the native CE registry to
+  // register stand-in elements that can delegate out to CE classes registered
+  // in scoped registries
+  class ShimmedCustomElementsRegistry implements CustomElementRegistry {
+    private readonly _definitionsByTag = new Map<
+      string,
+      CustomElementDefinition
+    >();
+    private readonly _definitionsByClass = new Map<
+      CustomElementConstructor,
+      CustomElementDefinition
+    >();
+    private readonly _whenDefinedPromises = new Map<
+      string,
+      AsyncInfo<CustomElementConstructor>
+    >();
+    private readonly _awaitingUpgrade = new Map<string, Set<HTMLElement>>();
+
+    define(tagName: string, elementClass: CustomElementConstructor) {
+      tagName = tagName.toLowerCase();
+      if (this._getDefinition(tagName) !== undefined) {
+        throw new DOMException(
+          `Failed to execute 'define' on 'CustomElementRegistry': the name "${tagName}" has already been used with this registry`
+        );
+      }
+      if (this._definitionsByClass.get(elementClass) !== undefined) {
+        throw new DOMException(
+          `Failed to execute 'define' on 'CustomElementRegistry': this constructor has already been used with this registry`
+        );
+      }
+      // Since observedAttributes can't change, we approximate it by patching
+      // set/remove/toggleAttribute on the user's class
+      const attributeChangedCallback =
+        elementClass.prototype.attributeChangedCallback;
+      const observedAttributes = new Set<string>(
+        elementClass.observedAttributes || []
       );
-    }
-    if (this._definitionsByClass.get(elementClass) !== undefined) {
-      throw new DOMException(
-        `Failed to execute 'define' on 'CustomElementRegistry': this constructor has already been used with this registry`
+      patchAttributes(
```

Comment:
There are a lot of what looks like superfluous formatting changes here. Can you revert those?

Reply:
I can, although this is as a result of indenting slightly more and running `npm run format`